include/realtime_tools/lock_free_queue.hpp

```diff
@@ -0,0 +1,358 @@
+   * @note Can be used in a multi threaded applications
+   */
+  template <
+    typename U, bool IsSPSCQueue = is_spsc_queue<LockFreeContainer>::value,
+    typename std::enable_if_t<!IsSPSCQueue, int> = 1>
+  [[nodiscard]] std::enable_if_t<std::is_convertible_v<T, U>, bool> bounded_push(const U & data)
+  {
+    if (!data_queue_.bounded_push(data)) {
+      T dummy;
+      data_queue_.pop(dummy);
+      return data_queue_.bounded_push(data);
+    }
+    return true;
+  }
+
+  /**
+   * @brief Check if the queue is empty
+   * @return true If the queue is empty
+   * @return false If the queue is not empty
+   * @note This function is enabled only if the queue is a spsc_queue
+   * @note Should only be called from the consumer thread where pop is called
+   */
+  template <
+    bool IsSPSCQueue = is_spsc_queue<LockFreeContainer>::value,
+    typename std::enable_if_t<IsSPSCQueue, int> = 0>
+  bool empty() const
+  {
+    return data_queue_.read_available() == 0;
+  }
+
+  /**
+   * @brief Check if the queue is empty
+   * @return true If the queue is empty
+   * @return false If the queue is not empty
+   * @note The result is only accurate, if no other thread modifies the queue.
+   * Therefore it is rarely practical to use this value in program logic.
+   * @note This function is enabled only if the queue is of multiple producer and multiple consumer type
+   */
+  template <
+    bool IsSPSCQueue = is_spsc_queue<LockFreeContainer>::value,
+    typename std::enable_if_t<!IsSPSCQueue, int> = 1>
+  bool empty() const
+  {
+    return data_queue_.empty();
+  }
+
+  /**
+   * @brief Get the capacity of the queue
+   * @return std::size_t Capacity of the queue
+   */
+  size_t capacity() const { return capacity_; }
+
+  /**
+   * @brief Get the size of the queue
+   * @return std::size_t Size of the queue
+   * @note This function is enabled only if the queue is a spsc_queue
+   */
+  template <
+    bool IsSPSCQueue = is_spsc_queue<LockFreeContainer>::value,
+    typename std::enable_if_t<IsSPSCQueue, int> = 0>
+  std::size_t size() const
+  {
+    return data_queue_.read_available();
+  }
+
+  /**
+   * @brief The method to check if the queue is lock free
+   * @return true If the queue is lock free, false otherwise
+   * @warning It only checks, if the queue head and tail nodes and the freelist can
+   * be modified in a lock-free manner. On most platforms, the whole implementation
+   * is lock-free, if this is true. Using c++0x-style atomics, there is no possibility
+   * to provide a completely accurate implementation, because one would need to test
+   * every internal node, which is impossible if further nodes will be allocated from
+   * the operating system.
+   * @link https://www.boost.org/doc/libs/1_74_0/doc/html/boost/lockfree/queue.html
+   */
+  bool is_lock_free() const
+  {
+    if constexpr (is_spsc_queue<LockFreeContainer>::value) {
+      return true;
+    } else {
+      return data_queue_.is_lock_free();
+    }
```

Comment:
```suggestion
    return constexpr (is_spsc_queue<LockFreeContainer>::value) || data_queue_.is_lock_free();
```

Reply:
I've removed the `constexpr` in your suggestions